modules/core/perf/opencl/perf_matop.cpp

```diff
@@ -13,6 +13,59 @@
 namespace cvtest {
 namespace ocl {
 
+
+///////////// oclCleanupCallback threadsafe check ////////////////////////
+
+typedef Size_MatType CleanupCallbackFixture;
+
+// Case 1: reuse of old src Mat in OCL pipe. Hard to catch!
+OCL_PERF_TEST(CleanupCallbackFixture, CleanupCallback_1)
+{
+    const Size srcSize = szQVGA;
+    const int type = CV_8UC1;
+    const int dtype = CV_16UC1;
+
+    checkDeviceMaxMemoryAllocSize(srcSize, dtype);
+
+    Mat src(srcSize, type);
+    UMat dst(srcSize, dtype);
+
+    declare.in(src, WARMUP_RNG).out(dst);
+
+    OCL_TEST_CYCLE_MULTIRUN(1000)
+    {
+        UMat tmpUMat = src.getUMat(ACCESS_RW);
+        tmpUMat.convertTo(dst, dtype);
+        ::cv::ocl::finish();
+    }
+
+    SANITY_CHECK(src);
```

Comment:
> This check requires test data about src values in opencv_extra repository. This check looks useless, so use SANITY_CHECK_NOTHING here.

Due to this data race there is scenario where src data become corrupted but no other fails happens, so I use this as additional check.

> Is there stable result (I mean time measurement for this performance test)? If no, please move tests to accuracy.

Ok.